### ISSUE TITLE
Possibility to parse/understand complex build outputs

### ISSUE THREAD

**Issue by sdaschner** [earthly/earthly#2841]
When building a project with many targets, sub-structures, or architectures, it's hard to get an understanding of the output and outcome of the individual targets, since the output is all over the place (similar what is described in #1525 and #1813).

Is it possible to get access to the build graph that has been created by buildkit after an earthly build? This can be in a very low-level, or technical format (JSON, or whatever is produced).

To understand better what could help us:

If we have a project that compiles, builds, and tests 100+ packages all in parallel (Earthfiles with imports and parameters/arch permutations), it would be nice to have some programmatic effort to, for example, grab the output of all `../*/+test` build targets, and see through it visually (or in a subsequent analysis).

Right now, all we can do is to manually scroll though the (huge) output and try to separate the log lines into the actual targets, which, as has been mentioned, are very messy.

Beside some new feature request, is there any chance to get some low-level access to the buildkit / graph?

**Comment by alexcb** [comment]
We've done some work on creating an internal [logbus](https://github.com/earthly/earthly/tree/main/logbus); which will eventually support such a feature.

Producing a graph is a bit more challenging, because the graph is **not** static, due to commands such as `IF`, and `FOR`, which must first execute part of the graph before it can continue generating the complete graph. A long term plan would be to evaluate these commands (whenever possible) within earthly rather than a container -- once we support that, then we could generate a complete graph prior to submitting it to buildkit.

**Comment by sdaschner** [comment]
Ok, that logbus approach sounds promising; is there already some way to early-access this via API or programmatic approach?

**Comment by alexcb** [comment]
The logbus code is not complete. It's hidden behind a feature flag, but is currently highly experimental and not supported.
